liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java

```diff
@@ -202,13 +209,22 @@ protected Plugin getPlugin(String groupId, String artifactId) {
     }
     
     protected Plugin getLibertyPlugin() {
+        // Try getting the version from Maven 3's plugin descriptor
+        String version = null;
+        if (plugin != null && plugin.getPlugin() != null) {
+            version = plugin.getVersion();
+            log.debug("Setting plugin version to " + version);
+        }
         Plugin plugin = project.getPlugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID + ":" + LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID);
         if (plugin == null) {
             plugin = getPluginFromPluginManagement(LIBERTY_MAVEN_PLUGIN_GROUP_ID, LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID);
         }
         if (plugin == null) {
             plugin = plugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID, LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID, "LATEST");
```

Comment:
Trying to think how we'd end up in this if-branch.. perhaps if someone else executed this via MojoExecutor ... I think that's what the Boost plugin is/was doing?

Reply:
I think it should always get into this if-branch.  Unless the user is on an old version of Maven (pre Maven 3) which does not have that `${plugin}` property as documented in https://maven.apache.org/ref/3.1.1/maven-core/apidocs/org/apache/maven/plugin/PluginParameterExpressionEvaluator.html.